packages/client/src/components/TransactionTable.js

```diff
@@ -58,25 +33,50 @@ const Row = ({ transaction, displayIndex, onView }) => {
 };
 
 const TransactionTable = ({ safeData, transactions = [], className = "" }) => {
-  const modalContext = useModalContext();
+  const { openModal, closeModal } = useModalContext();
 
   const onViewTransaction = (transaction) => {
-    modalContext.openModal(
-      <TransactionDetails
-        safeOwners={safeData.safeOwners}
-        transaction={transaction}
-        onClose={() => modalContext.closeModal()}
-      />
+    const { actionView, signerResponses } = transaction.blockEventData;
+    const signers = {};
+    safeData.safeOwners.forEach((owner) => {
+      let signerStatus = "pending";
+      if (signerResponses?.[owner.address] === 0) {
+        signerStatus = "approved";
+      } else if (signerResponses?.[owner.address] === 1) {
+        signerStatus = "rejected";
+      }
+      signers[owner.address] = signerStatus;
+    });
+    openModal(
+      <TransactionDetailsModal
+        onClose={closeModal}
+        safeData={safeData}
+        // massage data from graffle to work with TransactionDetails
+        transaction={{
+          status: transaction.status,
+          txID: transaction.flowTransactionId,
+          date: transaction.eventDate,
```

Comment:
prefer destructure before
```suggestion
 const {status, flowTransactionId, eventDate} = transaction
 ...
          status: status,
          txID: flowTransactionId,
          date: eventDate,
```